CRM/Report/Form.php

```diff
@@ -2637,23 +2643,22 @@ public function buildQuery($applyLimit = TRUE) {
    * Build group by clause.
    */
   public function groupBy() {
-    $groupBys = array();
     if (!empty($this->_params['group_bys']) &&
       is_array($this->_params['group_bys'])
     ) {
       foreach ($this->_columns as $tableName => $table) {
         if (array_key_exists('group_bys', $table)) {
           foreach ($table['group_bys'] as $fieldName => $field) {
             if (!empty($this->_params['group_bys'][$fieldName])) {
-              $groupBys[] = $field['dbAlias'];
+              $this->_groupByArray[] = $field['dbAlias'];
```

Comment:
I'm not too sharp on the overall flow of CiviReport classes, so humor me...

(1) The fix here is splitting apart `_groupBy` and `_groupByArray` as two variables? The old variable was doing too much (changing types somehow?), and new variable is more reliably typed?

(2) Would there ever be a way that `groupBy()` might run twice? (For example, once when building a query for a record-count, and again for a query that returns all data.) If it could get called twice, then wouldn't the content of `$this->_groupByArray` would be cumulative (e.g. possibly containing all the group-bys twice)?